File: test/src/test/java/com/github/variety/test/BasicAnalysisTest.java
package com.github.variety.test;

import com.github.variety.Variety;
import com.github.variety.validator.ResultsValidator;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

/**
 * Tests basic collection structure provided in readme of variety
 */
public class BasicAnalysisTest {

    private Variety variety;

    @Before
    public void setUp() throws Exception {
        this.variety = new Variety("test", "users");
        variety.getSourceCollection().insert(SampleData.getDocuments());
    }

    @After
    public void tearDown() throws Exception {
        variety.getVarietyResultsDatabase().dropDatabase();
        variety.getSourceCollection().drop();
    }

    /**
     * Validate correct results read from DB
     */
    @Test
    public void verifyBasicResultsDb() throws Exception {
        validate(variety.runDatabaseAnalysis());
    }

    /**
     * Validate correct results read from JSON standard output
     */
    @Test
    public void verifyBasicResultsJson() throws Exception {
        validate(variety.runJsonAnalysis());
    }

    private void validate(final ResultsValidator analysis) {
        analysis.validate("_id", 5, 100, "ObjectId");
        analysis.validate("name", 5, 100, "String");
        analysis.validate("bio", 3, 60, "String");
        analysis.validate("pets", 2, 40, "String", "Array");
        analysis.validate("someBinData", 1, 20, "BinData-old");
        analysis.validate("someWeirdLegacyKey", 1, 20, "String");
    }
}

File: spec/BasicAnalysisTest.js
import assert from 'assert';
import Tester from './utils/Tester.js';
const test = new Tester('test', 'users');

import sampleData from './assets/SampleData';
import expectedAscii  from './assets/ExpectedAscii';

describe('Basic Analysis', () => {

  beforeEach(() => test.init(sampleData));
  afterEach(() => test.cleanUp());

  it('should return ASCII results', async () => {
    const output = await test.runAnalysis({collection:'users'}, true);
    assert.equal(output, expectedAscii);
  });

  it('should return JSON results', async () => {
    const results = await test.runJsonAnalysis({collection:'users', lastValue:true}, true);
    results.validateResultsCount(7);
    results.validate('_id', 5, 100.0, {ObjectId: 5});
    results.validate('name', 5, 100.0, {String: 5}, 'Jim');
    results.validate('bio', 3, 60.0, {String: 3}, 'Ça va?');
    results.validate('birthday', 2,  40.0, {Date: 2}, 448070400000);
    results.validate('pets', 2,  40.0, {String: 1, Array: 1}, 'egret');
    results.validate('someBinData', 1,  20.0, {'BinData-generic': 1});
    results.validate('someWeirdLegacyKey', 1,  20.0, {String: 1}, 'I like Ike!');
  });
});
